app/services/network_cache/fetcher.rb

```diff
@@ -8,10 +8,12 @@ def self.from(testing_ground, opts = {})
 
     #
     # Fetches all cache and sets is as the load attribute for a node
-    def fetch
+    def fetch(nodes = [])
       tree_scope.each do |network|
         network.nodes.each do |node|
```

Comment:
... the `map.compact` step means this is probably not worth changing unless we do that validation of the request (to assert that the nodes named actually exist). Your call, but I'm not too bothered if this doesn't get changed.